Makefile

```diff
@@ -0,0 +1,105 @@
+# This Makefile captures common developer build and test use cases.
```

Comment:
I don't think we should mix make and cmake. @ihnorton do you have thoughts here? We don't mix things in any other repo.

Reply:
This idea came from SOMA, which does have a Makefile. It's a convenience Makefile to run common developer commands, so people don't have to come up with their own aliases, scripts, etc.